src/gui/dialogs/unit_recall.cpp

```diff
@@ -240,7 +247,7 @@ void unit_recall::pre_show(window& window)
 	list.register_sorting_option(0, [this](const int i) { return recall_list_[i]->type_name().str(); });
 	list.register_sorting_option(1, [this](const int i) { return recall_list_[i]->name().str(); });
 	list.register_sorting_option(2, [this](const int i) { return recall_list_[i]->level(); });
-	list.register_sorting_option(3, [this](const int i) { return recall_list_[i]->experience(); });
+	list.register_sorting_option(3, [this](const int i) { return recall_list_[i]->experience_to_advance(); });
```

Comment:
Please don't change how sorting by XP works. (This change is completely orthogonal to the target of sorting by both level and XP, as I explained.)